doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,247 @@
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of SubClientsClient for mocking. </summary>
+        protected SubClientsClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of SubClientsClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public SubClientsClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new SubClientsClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        private Parameter _cachedParameter;
+
+        /// <summary> Initializes a new instance of Root. </summary>
+        /// <param name="cachedParameter"> The String to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="cachedParameter"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="cachedParameter"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual Root GetRootClient(string cachedParameter)
+        {
+            Argument.AssertNotNullOrEmpty(cachedParameter, nameof(cachedParameter));
+
+            return new Root(ClientDiagnostics, _pipeline, _keyCredential, cachedParameter, _endpoint);
+        }
+
+        /// <summary> Initializes a new instance of Parameter. </summary>
+        public virtual Parameter GetParameterClient()
+        {
+            return Volatile.Read(ref _cachedParameter) ?? Interlocked.CompareExchange(ref _cachedParameter, new Parameter(ClientDiagnostics, _pipeline, _keyCredential, _endpoint), null) ?? _cachedParameter;
+        }
+    }
+}
+
+// SubClient: Root Generated\Root.cs
+namespace Azure.Service.SubClients
+{
+-   public partial class RootClient
++   public partial class Root
```

Comment:
Typically `root` means the root of a hierarchy, and I would expect the single top level client to have the name Root, not a subclient of it.  Is it possible to rename these to illustrate that?

Reply:
I updated to the Client1Client, and Client2Client to sample the clients in the PR